toplevel/metasyntax.ml

```diff
@@ -991,8 +992,8 @@ let compute_syntax_data df modifiers =
   let symbols' = remove_curly_brackets symbols in
   let need_squash = not (List.equal Notation.symbol_eq symbols symbols') in
   let ntn_for_grammar = make_notation_key symbols' in
-  check_rule_productivity symbols';
-  let msgs,n = find_precedence n etyps symbols' in
+  if not onlyprint then check_rule_productivity symbols';
+  let msgs,n = if onlyprint then ([],0) else find_precedence n etyps symbols' in
```

Comment:
The precedence matters for printing, so that no extra parentheses are added, and so that no needed parentheses are missing. So it should be computed even if only for printing. However, the given values seem ok to me for the empty string (`[]` to mean no messages to display, `0` to mean printed at level 0, which is the lower level, so that no parentheses should be printed around the empty string). So, maybe `find_precedence` can do the test of non-productivity and return `([],0)` itself. The test could be for instance a list of symbols made only of `Break`, as tested in the code computing `first_symbol`.

Looks nice otherwise!